### PR TITLE
Use ChecksumAlgorithm property

### PR DESCRIPTION
**Pull request by tmat**
csc task uses `ChecksumAlgorithm` property to set the checksum algorithm. No need for .rsp file.

### CODE REVIEW

**Comment by tmat** [comment]
@weshaggard

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.props [hunk 1]
```diff
@@ -16,7 +16,7 @@
   <Import Project="$(MSBuildThisFileDirectory)RepoAPI.Mapping.props" />
 
   <PropertyGroup>
-    <CompilerResponseFile Condition="'$(CheckSumSHA256)'!='false'">$(MSBuildThisFileDirectory)checksum.rsp</CompilerResponseFile>
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.props]: I'm not sure if anyone is setting the condition but could you add the same condition to this property?

**Comment by weshaggard** [comment]
cc @joperezr

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.props (hunk 1 above)
Comment (tmat) [reply src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.props]: I'd rather not add a condition that no one uses. If there is any scenario that requires SHA1 then we need to know about it and make it not require SHA1. If anything gets broken `ChecksumAlgorithm` can be set directly to unblock such scenario temporarily.

Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.props]: The previous task library we were using didn't support this argument, which is why we needed to pass in a response file instead. Now that we are using the updated task library we can make use of it.

Comment (weshaggard) [reply src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.props]: >I'd rather not add a condition that no one uses. If there is any scenario that requires SHA1 then we need to know about it and make it not require SHA1. If anything gets broken ChecksumAlgorithm can be set directly to unblock such scenario temporarily.

Sounds good and your right there is a temp fallback if needed.

**Pull request merged**